manifests/provider/systemd.pp

```diff
@@ -0,0 +1,53 @@
+#
+# @summary manages a systemd wireguard interface
+#
+# @param interface the name for the wg interface
+# @param peers is an array of struct (Wireguard::Peers) for multiple peers
+# @param dport destination for firewall rules / where our wg instance will listen on. defaults to the last digits from the title
+# @param addresses different addresses for the systemd-networkd configuration
+# @param description an optional string that will be added to the wireguard network interface
+# @param mtu configure the MTU (maximum transision unit) for the wireguard tunnel. By default linux will figure this out. You might need to lower it if you're connection through a DSL line. MTU needs to be equal on both tunnel endpoints
+# @param routes different routes for the systemd-networkd configuration
+# @param preshared_key Define preshared key which should be used for this interface
```

Comment:
This class look like an implementation detail not intended for end-users.

We can flag this at the documentation level:

```puppet
# @api private
```

and at puppet level thanks to assert_private from stdlib:

```puppet

class wireguard::provider::systemd (
  # ...
) {
  assert_private
```

Reply:
awesome, i didn't know about this. thanks!